worknenjoy/gitpay
# Create Report Issue Button

## Description

> Created a _Report_ button on the task page that notifies the email _report@gitpay.me_ that there is an issue with the task.

## Changes

- Frontend: added a report button that opens a form for users to send a report based on a selected reason. Created a redux action reportTask that makes a call to the server once the form is sent.
- Backend: Created a route that sends an email to _report@gitpay.me_ and a route to delete the the task that is sent as a url in the report email. Once a task is reported a token is created and attached as a query parameter to the delete route to secure it. 

## Issue

> Closes #683 

## Impacted Area

> https://gitpay.me/#/task/taskId

## Steps to test

Steps needed to reproduce the scenario from this change to validate if the pull request solve this issue

- Create a task with an account
- Log into a different account and report the created task
- _report@gitpay.me_ should receive email to delete the task
- Check the email of the account that created that task to confirm that an email notifying them of the task deletion was sent.

## Before
> Screenshot from the state before

![Screen Shot 2020-07-16 at 10 09 41 PM](https://user-images.githubusercontent.com/44875087/87740793-185b5900-c7b1-11ea-84c8-8b9dbfe6377c.png)

## After
> Screenshot from the state after your Pull Request

![Screen Shot 2020-07-16 at 10 11 12 PM](https://user-images.githubusercontent.com/44875087/87740884-4b055180-c7b1-11ea-9651-54dcb3e74f89.png)

![Screen Shot 2020-07-16 at 10 12 08 PM](https://user-images.githubusercontent.com/44875087/87740950-683a2000-c7b1-11ea-85d6-81f65828fc5f.png)
